src/Simplex/Messaging/Agent/Store/SQLite.hs

```diff
@@ -41,51 +42,70 @@ import Database.SQLite.Simple.ToField (ToField (..))
 import Network.Socket (ServiceName)
 import Simplex.Messaging.Agent.Protocol
 import Simplex.Messaging.Agent.Store
-import Simplex.Messaging.Agent.Store.SQLite.Schema (createSchema)
+import qualified Simplex.Messaging.Agent.Store.SQLite.Migrations as Migrations
 import Simplex.Messaging.Parsers (blobFieldParser)
 import qualified Simplex.Messaging.Protocol as SMP
 import Simplex.Messaging.Util (bshow, liftIOEither)
-import System.Exit (ExitCode (ExitFailure), exitWith)
+import System.Directory (copyFile, createDirectoryIfMissing, doesFileExist)
+import System.Exit (exitFailure)
 import System.FilePath (takeDirectory)
+import System.IO (hFlush, stdout)
 import Text.Read (readMaybe)
-import UnliftIO.Directory (createDirectoryIfMissing)
 import qualified UnliftIO.Exception as E
 
 -- * SQLite Store implementation
 
 data SQLiteStore = SQLiteStore
   { dbFilePath :: FilePath,
-    dbConn :: DB.Connection
+    dbConn :: DB.Connection,
+    dbNew :: Bool
   }
 
-createSQLiteStore :: MonadUnliftIO m => FilePath -> m SQLiteStore
+createSQLiteStore :: FilePath -> IO SQLiteStore
 createSQLiteStore dbFilePath = do
   let dbDir = takeDirectory dbFilePath
   createDirectoryIfMissing False dbDir
   store <- connectSQLiteStore dbFilePath
-  compileOptions <- liftIO (DB.query_ (dbConn store) "pragma COMPILE_OPTIONS;" :: IO [[T.Text]])
+  compileOptions <- DB.query_ (dbConn store) "pragma COMPILE_OPTIONS;" :: IO [[T.Text]]
   let threadsafeOption = find (isPrefixOf "THREADSAFE=") (concat compileOptions)
-  liftIO $ case threadsafeOption of
-    Just "THREADSAFE=0" -> do
-      putStrLn "SQLite compiled with not threadsafe code, continue (y/n):"
-      s <- getLine
-      when (s /= "y") (exitWith $ ExitFailure 2)
+  case threadsafeOption of
+    Just "THREADSAFE=0" -> confirmOrExit "SQLite compiled with non-threadsafe code."
     Nothing -> putStrLn "Warning: SQLite THREADSAFE compile option not found"
     _ -> return ()
-  liftIO . createSchema $ dbConn store
-  return store
-
-connectSQLiteStore :: MonadUnliftIO m => FilePath -> m SQLiteStore
+  migrateSchema store
+  pure store
+
+migrateSchema :: SQLiteStore -> IO ()
+migrateSchema SQLiteStore {dbConn, dbFilePath, dbNew} = do
+  Migrations.initialize dbConn
+  Migrations.get dbConn Migrations.app >>= \case
+    Left e -> confirmOrExit $ "Database error: " <> e
+    Right [] -> pure ()
+    Right ms -> do
+      unless dbNew $ do
+        confirmOrExit "The app has a newer version than the database - it will be backed up and upgraded."
```

Comment:
and re-phrase this please - the second part now ambiguously reads as if it's the app that is backed up and upgraded (which is ofc total nonsense from the context).
either:
`The app has a newer version than the database - database will be backed up and upgraded.`
or
`The database has an older version than the app - it will be backed up and upgraded.`

Reply:
"it" can bind both the subject and the object, and it is usually thought of as binding the nearest, which is the database... Also you cannot backup the app anyway.